internal/resources/provisioner/data_source_provisioner.go

```diff
@@ -28,35 +31,48 @@ func DataSourceProvisioner() *schema.Resource {
 }
 
 var provisionerListSchema = map[string]*schema.Schema{
-	nameKey: {
-		Type:        schema.TypeString,
-		Description: "Name of the provisioner",
-		Optional:    true,
-	},
-	managementClusterNameKey: {
-		Type:        schema.TypeString,
-		Description: "Name of the management cluster",
-		Required:    true,
-		ForceNew:    true,
-	},
-	orgIDKey: {
-		Type:        schema.TypeString,
-		Description: "ID of the organization",
+	provisionerKey: {
+		Type:        schema.TypeList,
+		Description: "Provisioners info",
 		Optional:    true,
+		Elem: &schema.Resource{
+			Schema: map[string]*schema.Schema{
+				nameKey: {
+					Type:        schema.TypeString,
+					Description: "Name of the provisioner",
+					Optional:    true,
+				},
+				managementClusterNameKey: {
+					Type:        schema.TypeString,
+					Description: "Name of the management cluster",
+					Required:    true,
+					ForceNew:    true,
+				},
+				orgIDKey: {
+					Type:        schema.TypeString,
+					Description: "ID of the organization",
+					Optional:    true,
+				},
+				common.MetaKey: common.Meta,
+			},
+		},
 	},
-	common.MetaKey: common.Meta,
 }
 
 func dataSourceProvisionerRead(ctx context.Context, d *schema.ResourceData, m interface{}) (diags diag.Diagnostics) {
 	config := m.(authctx.TanzuContext)
 
-	model, err := tfModelConverter.ConvertTFSchemaToAPIModel(d, []string{nameKey, managementClusterNameKey})
-	if err != nil {
+	id := make([]string, 0)
+
+	var resp *provisioner.VmwareTanzuManageV1alpha1ManagementclusterProvisionerListprovisionersResponse
+
+	model, err := tfModelDataConverter.ConvertTFSchemaToAPIModel(d, []string{provisionerKey, nameKey, managementClusterNameKey})
+	if err != nil || model == nil {
 		return diag.FromErr(errors.Wrapf(err, "Couldn't read Tanzu Mission Control provisioner configurations."))
 	}
 
-	if model.FullName.Name == "" {
-		resp, err := config.TMCConnection.ProvisionerResourceService.ProvisionerResourceServiceList(model.FullName)
+	if model.Provisioners[0].FullName.Name == "" {
```

Comment:
before accessing the element in array like `model.Provisioners[0]`, add a check to validate list is not empty.

Reply:
Have added the check to see if the model is nil or not in line 70

Reply:
better to add  a nil check for provisioner too `model.Provisioners == nil`